app/server/middleware/search.js

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const searchArticles = (request, response, next) => {
+  // only searches a single source, rather than an
+  // array of sources
+  // also doesn't allow sortBy filtering yet, as some
+  // sources don't support all types of filtering
+  const { source } = request.query;
+
+  const search = `https://newsapi.org/v1/articles?apiKey=${process.env.NEWS_KEY}&source=${source}`;
+
+  axios.get(search)
+    .then((data) => {
+      const articles = data.data.articles.map((article) => {
```

Comment:
`data.data` seems weird to me.  maybe think about a more intuitive naming?

Reply:
Sure--I've updated the data parameter to newsResponse (so as not to mix it up with our server's response to the client), so this now reads as newsResponse.data.articles.